client/client.go

```diff
@@ -752,3 +756,41 @@ type HistoryJSONOptions struct {
 func HistoryFromJSON(r io.Reader, options HistoryJSONOptions) (*historypb.History, error) {
 	return internal.HistoryFromJSON(r, options.LastEventID)
 }
+
+// NewAPIKeyCredentials creates credentials that can be provided to
+// ClientOptions to use a fixed API key.
+//
+// This is the equivalent of providing a headers provider that sets the
+// "Authorization" header with "Bearer " + the given key. This will overwrite
+// any "Authorization" header that may be on the context or from existing header
+// provider.
+//
+// Note, this uses a fixed header value for authentication. Many users that want
+// to rotate this value without reconnecting should use
+// [NewAPIKeyCallbackCredentials].
+func NewAPIKeyCredentials(apiKey string) Credentials {
```

Comment:
Wonder if it's worth calling this `NewAPIKeyStaticCredentials` and the callback based one `NewAPIKeyDynamicCredentials`. Not critical, just thought we may want to consider alternative names.

Reply:
I don't mind alternative names, these work for me. Will leave this discussion open for others before changing.

Reply:
+1 to calling this static. Callback one I think can stay as is but I see Roey's name as good too. Either way.

Reply:
Updated names